File: java/src/main/videostore/Movie.java
package videostore;

public class Movie {
    public static final int CHILDREN = 2;
    public static final int REGULAR = 0;
    public static final int NEW_RELEASE = 1;
    private String _title;
    private Price _price;

    public Movie(String title, int priceCode) {
        _title = title;
        setPriceCode(priceCode);
    }

    public int getPriceCode() {
        return _price.getPriceCode();
    }

    public void setPriceCode(int arg) {
        switch (arg) {
            case REGULAR:
                _price = new RegularPrice();
                break;
            case NEW_RELEASE:
                _price = new NewReleasePrice();
                break;
            case CHILDREN:
                _price = new ChildrenPrice();
                break;
            default:
                throw new IllegalArgumentException("The price code is wrong.\n");
        }
    }

    public String getTitle() {
        return _title;
    }

    double getCharge(int dayRented) {
        return _price.getCharge(dayRented);
    }

    int getFrequentRenterPoints(int daysRented) {
        return _price.getFrequentRenterPoints(daysRented);
    }
}

File: python/movie.py
from price import *


class Movie:

    def __init__(self, title, price_code):
        self._title = title
        self._price = Price()
        self.price_code(price_code)

    @property
    def price_code(self):
        return self._price.price_code

    def price_code(self, price_code):
        if price_code == MovieCode.REGULAR:
            self._price = RegularPrice()
        elif price_code == MovieCode.NEW_RELEASE:
            self._price = NewReleasePrice()
        elif price_code == MovieCode.CHILDREN:
            self._price = ChildrenPrice()

    @property
    def title(self):
        return self._title

    def cost(self, days_rented):
        return self._price.cost(days_rented)

    def points(self, days_rented):
        return self._price.points(days_rented)
